### PR TITLE
Update install instructions

### PR DESCRIPTION
**Pull request by tpdorsey**
Updates installation instructions for v1 after v2 GA.

Replaces https://github.com/PrefectHQ/prefect/pull/6047

### CODE REVIEW

**Review by chrisguidry — APPROVED** [review]
That definitely works, is a valid PEP 440 version specifier (and does work, I tested it).  It might be a _little_ more conventional to say `pip install "prefect>=1.2.4,<2"`, but it's also uglier that way.

**Review by billpalombi — APPROVED** [review]
lgtm

**Comment by tpdorsey** [comment]
@desertaxle I'm getting failing tests on this related to task library. Any thoughts what's going on here?

**Comment by desertaxle** [comment]
Looks like one of the tasks had a client change that's breaking the tests. I'll submit a PR for a fix to unblock the pipeline.

**Comment by desertaxle** [comment]
Pulling in the changes from #6153 should fix the failing tests

File: docs/core/getting_started/install.md [hunk 1]
```diff
@@ -4,29 +4,33 @@
...
+pip install "prefect==1.*"
 ```
 
 :::
 
 ::: tab Conda
 
 ```bash
-conda install -c conda-forge prefect
+conda install -c conda-forge "prefect>=1.2.4,<2"
```
Comment (madkinsz) [inline docs/core/getting_started/install.md:25]: Why is the `>=` necessary?

Comment (madkinsz) [reply docs/core/getting_started/install.md:25]: I don't think we want to have a specific lower pin

Comment (tpdorsey) [reply docs/core/getting_started/install.md:25]: What should this line be?

**Pull request merged**